### PR TITLE
test(e2e,playwright): certification page

### PR DESCRIPTION
**Pull request by shabeebk**
With this commit, the playwright will test the certification page by iterating through the wholse set of certifications available using the slug and checks for title and other info.

Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Ref #51705

### CODE REVIEW

**Comment by shabeebk** [comment]
camperbot cant code

**Comment by shabeebk** [comment]
Every certificate slug will be checked against generic identical tests. This can further be enhanced to incorporate subjective checks to test against specific cases like the Microsoft logo.

Screenshot 2023-10-09 at 13 21 04

File: e2e/certification.spec.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  // currently live, but are still in the codebase.
+  if (blackListedCerts.includes(certificate)) {
+    continue;
+  }
+
+  test.describe(`Certification page -> test for ${certificate}`, () => {
+    test.beforeAll(async ({ browser }) => {
+      page = await browser.newPage();
+      await page.goto(`/certification/certifieduser/${certificate}`);
+    });
```
Comment (raditotev) [inline e2e/certification.spec.ts:31]: This is considered an anti-pattern. See more [here](https://github.com/freeCodeCamp/freeCodeCamp/pull/51754#discussion_r1342543051).

**Comment by shabeebk** [comment]
Thanks for those remarks and suggestions. I will push the additional commit to fix and comment.

File: e2e/certification.spec.ts (hunk 1 above)
Comment (shabeebk) [reply e2e/certification.spec.ts]: Updated and pushed.

**Comment by shabeebk** [comment]
Updated and pushed new commit.

**Comment by shabeebk** [comment]
Screenshot 2023-10-11 at 9 34 45

File: e2e/certification.spec.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+test.describe('Certification page', () => {
+  test('The certificate page has a donation section', async ({ page }) => {
+    const donation = page.getByTestId('donation-section');
+    await expect(donation).toBeVisible();
+
+    const donationText = page.getByTestId('donation-text');
+    await expect(donationText).toHaveText(
+      `Only you can see this message. Congratulations on earning this certification. It's no easy task. Running freeCodeCamp isn't easy either. Nor is it cheap. Help us help you and many other people around the world. Make a tax-deductible supporting donation to our charity today.`
+    );
```
Comment (raditotev) [inline e2e/certification.spec.ts:17]: It's better if you get the text from the translations: `client/i18n/locales/english/translations.json`

File: e2e/certification.spec.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+    const twitterLink = page.getByTestId('twitter-share-btn');
+    await expect(twitterLink).toBeVisible();
+    await expect(twitterLink).toHaveAttribute(
+      'href',
+      'https://twitter.com/intent/tweet?text=I just earned the Responsive%20Web%20Design certification @freeCodeCamp! Check it out here: https://freecodecamp.org/certification/certifieduser/responsive-web-design'
+    );
+
+    await expect(page.getByTestId('project-links')).toBeVisible();
```
Comment (raditotev) [inline e2e/certification.spec.ts:65]: Can you extend the test here with assertions for each of the links?

**Review by raditotev — COMMENTED** [review]
It will be nice if you include test case for Microsoft certificate.

**Comment by shabeebk** [comment]
@raditotev So, do you think we should test both cases, or testing against Microsoft is good enough? I could see that Microsoft has an extra logo and signee.

**Comment by raditotev** [comment]
I think both cases should be covered.

File: e2e/certification.spec.ts (hunk 2 above)
Comment (shabeebk) [reply e2e/certification.spec.ts]: Fixed

File: e2e/certification.spec.ts (hunk 3 above)
Comment (shabeebk) [reply e2e/certification.spec.ts:164]: Should I add the tests for all those links?

**Review by naomi-lgbt — APPROVED** [review]

File: e2e/certification.spec.ts [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+    const certWrapper = page.getByTestId('cert-wrapper');
+    await expect(certWrapper).toBeVisible();
+
+    const header = certWrapper.getByTestId('cert-header');
+    await expect(header).toBeVisible();
+    const fccLogo = header.getByTestId('cert-fcc-logo');
+    await expect(fccLogo).toBeVisible();
+    // Non Microsoft certs should not have this one
+    const microsoftLogo = header.getByTestId('cert-microsoft-logo');
+    await expect(microsoftLogo).not.toBeVisible();
```
Comment (raditotev) [inline e2e/certification.spec.ts:32]: Nice!

File: e2e/certification.spec.ts [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+
+    const certInfoContainer = certWrapper.getByTestId('cert-info-container');
+    await expect(certInfoContainer).toBeVisible();
+    const certTitle = certInfoContainer.getByTestId('certification-title');
+    await expect(certTitle).toHaveText(
+      translations.certification.title['Responsive Web Design']
```
Comment (raditotev) [inline e2e/certification.spec.ts:38]: 👍

**Review by raditotev — APPROVED** [review]
Thank you for your contribution to the page! 👍
We are happy to accept these changes and look forward to future contributions. 🎉

**Pull request merged**